featurebyte/service/offline_store_feature_table.py

```diff
@@ -50,6 +75,55 @@ async def get_or_create_document(
             return OfflineStoreFeatureTableModel(**query_result["data"][0])
         return await self.create_document(data)
 
+    async def _populate_remote_attributes(
+        self, document: OfflineStoreFeatureTableModel
+    ) -> OfflineStoreFeatureTableModel:
+        if document.feature_cluster_path:
+            feature_cluster_json = await self.storage.get_text(Path(document.feature_cluster_path))
+            feature_cluster_dict = json_util.loads(feature_cluster_json)
+            document.feature_cluster = FeatureCluster(**feature_cluster_dict)
+        return document
+
+    async def _move_feature_cluster_to_storage(
+        self, document: OfflineStoreFeatureTableModel
+    ) -> OfflineStoreFeatureTableModel:
+        feature_cluster_path = self.get_full_remote_file_path(
+            f"offline_store_feature_table/{document.id}/feature_cluster.json"
+        )
+        assert isinstance(document.feature_cluster, FeatureCluster)
+        await self.storage.put_text(
+            json_util.dumps(document.feature_cluster.json_dict()), feature_cluster_path
+        )
+        document.feature_cluster_path = str(feature_cluster_path)
+        document.feature_cluster = None
+        return document
+
+    async def _create_document(
+        self, data: OfflineStoreFeatureTableModel
+    ) -> OfflineStoreFeatureTableModel:
+        if data.entity_lookup_info is None:
+            # check if name already exists
+            data.name = data.get_name()
+            query_filter = {"name": data.name}
+            query_result = await self.list_documents_as_dict(query_filter=query_filter, page_size=1)
+
+            count = query_result["total"]
+            data.name_suffix = None
+            if count:
+                # if name already exists, append a suffix
+                data.name_suffix = str(count)
+                data.name = data.get_name()
+
+        data = await self._move_feature_cluster_to_storage(data)
+        try:
+            output = await super().create_document(data)
+            assert output.catalog_id == data.catalog_id
+            return output
+        except Exception as exc:
+            if data.feature_cluster_path:
+                await self.storage.delete(Path(data.feature_cluster_path))
```

Comment:
Should this use `try_delete_if_exists`?

Reply:
On the second thought, i think it should be ok not to use `try_delete_if_exists` here. When this point is reached, it means the remote file has already been created (see line 117).